semver.py

```diff
@@ -261,6 +257,13 @@ def _compare_by_keys(d1, d2):
     return rccmp
 
 
+def _ensure_is_comparable(other):
+    comparable_types = (VersionInfo, dict, tuple)
```

Comment:
@ppkt made a good point about using interfaces in place of dict/tuple. This will allow us to compare with dict-like / tuple-like objects for free (well, not really, but nothing to care about).

The only thing is `str` - it's Iterable, but unlikely `*str` will give us what we really expected from it.